### ISSUE TITLE
Can't build on/for alpine linux on v1.1.0

### ISSUE THREAD

**Issue by nenkoru** [openziti/ziti-tunnel-sdk-c#931]
Trying to build a ziti-tunneler for alpine linux and getting an issue.

env:
```
Linux buildkitsandbox 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 GNU/Linux
PRETTY_NAME="Alpine Linux v3.20"
ninja version: 1.9                                                                      
gcc version 13.3.0 (GCC)

```
Build instructions for cmake:
```
git --git-dir=/ziti-tunnel-sdk-c/.git --work-tree=/ziti-tunnel-sdk-c checkout v1.1.0

cmake -DDISABLE_LIBSYSTEMD_FEATURE=ON -DCMAKE_C_COMPILER=/toolchain/bin/gcc -DCMAKE_CXX_COMPILER=/toolchain/bin/g++ -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja -DCMAKE_BUILD_TYPE=Release --preset ci-linux-x64 -S /ziti-tunnel-sdk-c -B /ziti-tunnel-sdk-c/build

cmake --build /ziti-tunnel-sdk-c/build # errors here
```

Part of the `cmake --build` log that causes an issue:
```
[153/176] Building C object _deps/tlsuv-build/CMakeFiles/tlsuv.dir/Debug/src/compression.c.o
1.941 ninja: job failed: /toolchain/bin/gcc -DSODIUM_STATIC -DZITI_BRANCH=HEAD -DZITI_COMMIT=g9fa5162 -DZITI_LOG_MODULE=\"ziti-sdk\" -DZITI_VERSION=1.0.7 -DCMAKE_INTDIR=\"Debug\" -I/ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/../includes -I/ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/../inc_internal -I/ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/proto -I/ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-build/include -I/ziti-tunnel-sdk-c/build/_deps/tlsuv-src/src -I/ziti-tunnel-sdk-c/build/_deps/tlsuv-src/include -I/ziti-tunnel-sdk-c/build/_deps/tlsuv-src/deps/uv_link_t/include -isystem /ziti-tunnel-sdk-c/build/vcpkg_installed/x64-linux/include -isystem /ziti-tunnel-sdk-c/build/vcpkg_installed/x64-linux/include/json-c -isystem /ziti-tunnel-sdk-c/build/vcpkg_installed/x64-linux/lib/pkgconfig/../../include -g -std=gnu11 -fPIC -g -pthread -MD -MT _deps/ziti-sdk-c-build/library/CMakeFiles/ziti.dir/Debug/internal_model.c.o -MF _deps/ziti-sdk-c-build/library/CMakeFiles/ziti.dir/Debug/internal_model.c.o.d -o _deps/ziti-sdk-c-build/library/CMakeFiles/ziti.dir/Debug/internal_model.c.o -c /ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/internal_model.c
1.941 In file included from /ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/../inc_internal/internal_model.h:19,
1.941                  from /ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/internal_model.c:16:
1.941 /ziti-tunnel-sdk-c/build/_deps/ziti-sdk-c-src/library/../inc_internal/internal_model.h:52:4: error: field 'expires' has incomplete type
1.941    52 | XX(expires, timestamp, none, expiresAt, __VA_ARGS__) \
```

**Comment by nenkoru** [comment]
Out of ideas atm on how to remedy this problem.
Any push to the direction would be helpful!

**Comment by scareything** [comment]
Hi and thanks for the report. Some changes to one of our dependencies happened to rely on struct timeval being defined. I made the relevant fix in the dependency project. Once that change is merged and released I'll make another PR in this project to use the updated dependency. Stand by.

**Comment by nenkoru** [comment]
Wow, thats fast.
Going to check the build in a few hours against v1.1.2.
Thanks!!

**Comment by nenkoru** [comment]
Hooray! It works as expected. Build is working fine now.
Closing the issue!
